### ISSUE TITLE
Improve a11y on SVGs throughout Gutenberg

### ISSUE THREAD

**Issue by ndiego** [WordPress/gutenberg#38286]
### Description

This issue stems from the work done in #38152 

In many places throughout Gutenberg, we render SVGs with `role="img"` but we do not include an `aria-label`. At first glance this may seem ok because in all instances we also have something like this:

```
<svg role="img" aria-hidden="true" focusable="false" width="24" height="24" viewBox="0 0 24 24" version="1.1" xmlns="http://www.w3.org/2000/svg">
      ...
</svg>
```

However in my reading about a11y, I don't believe `aria-hidden="true" focusable="false"` is sufficient here. Since we are including `role="img"`, there should also be an `aria-label` as described here in the [MDN Wed Docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/img_role#svg_and_roleimg). Now I am by no means a a11y expert, so I would love someone else to confirm or deny this issue. 

If it is an issue, the solution appears to be very simple. We just remove `role="img"` from all these SVGs since they are intended to be hidden from screen readers anyway. They are purely decoration and/or have other descriptive text that is readable by screen readers. 

If someone can confirm, I am happy to put together this PR. 

### Offending files/blocks: 

- [Navigation Link](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/navigation-link/index.php)
- [Navigation Submenu](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/navigation-submenu/index.php)
- [Navigation](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/navigation/index.php)
- [Page List](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/page-list/index.php)
- [Social Link](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/src/social-link)

### Step-by-step reproduction instructions

1. Using the latest version of Gutenberg, add a Social Icons block to the page.
2. Add a couple social links
3. Preview the page on the frontend and notice `role="img"` being added to all SVG icons.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by aurooba** [comment]
@alexstine do you have thoughts on this?

**Comment by alexstine** [comment]
Maybe just remove:
```
role="img"
```
and call it done?

CC @joedolson

**Comment by ndiego** [comment]
☝️ Yeah that was my thinking

**Comment by ndiego** [comment]
From the #accessibility channel on WordPress slack:

> If  you use role "presentation", you can remove aria-hidden="true", since the SVG usually doesn't have child content anyway (at least for a11y), and role "presentation" removes it from the tree, too. Or you remove the role alltogether and just leave aria-hidden in, that would also work. - marcoz

The general concensus seems to be that removing `role="img"` is a perfectly valid solution to this issue. I will work on a PR for this later today.

**Comment by kraftner** [comment]
I've recently come across [this tweet](https://twitter.com/mmatuzo/status/1481989095018614785) by @matuzo which hints at an issue with `role=presentation` with VO in Safari which might be relevant here.

**Comment by joedolson** [comment]
`<use>` has a native role of image, and `role=presentation` doesn't cascade (at least, not normally - there are exceptions, such as tables). aria-hidden will override the `use` element, however, as it removes the element and its children from the accessibility tree. 

This mostly has to do with the fact that, unlike `img`, an `svg` is a container with descendants, so you could end up in a situation where you want the parent svg to be treated presentationally, but have the children read individually.

**Comment by matuzo** [comment]
> I've recently come across [this tweet](https://twitter.com/mmatuzo/status/1481989095018614785) by @matuzo which hints at an issue with `role=presentation` with VO in Safari which might be relevant here.

There was some follow up discussion in the a11y slack channel and people there said pretty much what @joedolson explained in his reply. I've never used "role=presentation" for remove images from the accessibility tree (because generally that's not what it's supposed to do) before, but I see it on many sites I audit.
